optuna/integration/botorch.py

```diff
@@ -36,7 +36,19 @@
     from botorch.models import SingleTaskGP
     from botorch.models.transforms.outcome import Standardize
     from botorch.optim import optimize_acqf
-    from botorch.sampling.samplers import SobolQMCNormalSampler
+
+    try:
+        from botorch.sampling.normal import SobolQMCNormalSampler
+
+        def _get_sobol_qmc_normal_sampler(num_samples: int) -> SobolQMCNormalSampler:
+            return SobolQMCNormalSampler(sample_shape=torch.Size((num_samples,)))
+
+    except ImportError:
+        from botorch.sampling.samplers import SobolQMCNormalSampler
+
+        def _get_sobol_qmc_normal_sampler(num_samples: int) -> SobolQMCNormalSampler:
+            return SobolQMCNormalSampler(num_samples=num_samples)
+
```

Comment:
```suggestion
    from botorch.sampling import SobolQMCNormalSampler

    def _get_sobol_qmc_normal_sampler(num_samples: int) -> SobolQMCNormalSampler:
        return SobolQMCNormalSampler(num_samples)  # type: ignore
```

Reply:
I checked this to pass `tests/integration_tests/test_botorch.py` on botorch versions, 0.7.0 and 0.8.1.

Reply:
But this behavior will soon be deprecated, so I think it is better to do conditional branching based on the botorch versions.

Reply:
Thank you for investigations!

Reply:
I think we could delay adding a conditional until it is actually deprecated, if it is not currently so. @HideakiImamura What do you think?

Reply:
`botorch` would emit a [warning](https://github.com/pytorch/botorch/blob/main/botorch/sampling/base.py#L66-L71).

Reply:
I see, I fixed it.